src/services/nep141StorageService.ts

```diff
@@ -17,33 +15,30 @@ export type Output =
 
 /**
  * Get the amount of NEP-141 storage required for the user to store the token.
- * @param token The token to check.
+ * @param tokenAddress.
+ * @param tokenChainName.
  * @param userAccountId The user's NEAR account ID.
  * @returns The amount of NEAR required for the user to store the token; 0 means no storage required.
  */
 export async function getNEP141StorageRequired({
-  token,
+  tokenAddress,
   userAccountId,
 }: {
-  token: BaseTokenInfo
+  tokenAddress: string
   userAccountId: string
 }): Promise<Output> {
-  if (token.chainName !== "near" || isNativeToken(token)) {
```

Comment:
I believe we should not have this in this function. It makes it is hard to reason about it and introduces unnecessary complexity, which can be avoided easily.

Reply:
So you just moved this logic somewhere else, you didn't reduce complexity.

Reply:
the purpose for this function in my opinion should be to deal with only nep141 and nothing else, why the function should be aware of any other chains beside NEAR ?

Reply:
Having this function reason about diff chains deff not making the whole thing simpler

Reply:
To have this function accepting token like before it would also force to make us have all sorts of weird checks to handle the typing

Reply:
I would rather have a different function all together that accepts an abstract token and says if storage_deposit is even needed

Reply:
So you just want to create another function with the same logic. Current function is not that big to have concerns about it.